test/files/run/t12569.scala

```diff
@@ -0,0 +1,18 @@
+
+final class C[A, B](pf: PartialFunction[A, B]) {
+  def f(x: A) =
+    pf.applyOrElse(x, C.fallback) match {
+      case _: C.fallback.type => -1
+      case _ => 42
+    }
+}
+object C {
+  val fallback: Any => Any = _ => fallback
+}
```

Comment:
Cool, this will work with 

```scala

object C extends (Any => Any) {
  def apply(x:Any) ：Any = this
}
```
too right?

Reply:
It's just an experiment that skips checkability if no type tests are expected. But yes, the idea is to skip the check for singleton match of function value. Hopefully other checks will still work. No idea yet if this is a ballpark good idea.